cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -47,7 +61,11 @@ virsh destroy $DOMAIN_NAME || true
 virt-sysprep -d $DOMAIN_NAME --operations machine-id,bash-history,logfiles,tmp-files,net-hostname,net-hwaddr
 
 # Remove VM
-virsh undefine $DOMAIN_NAME
+if [[ ${ARCH} = "aarch64" ]]; then
+  virsh undefine --nvram "${DOMAIN_NAME}" || true
+else
+  virsh undefine "${DOMAIN_NAME}" || true
+fi
```

Comment:
Why is nvram flag is necessary?

If its mandatory please consider extracting this to function, it will make it easier to follow the script, something like:
```
function undefine_vm() {
    local -r domain=$1
    local -r arch=$2

    if [[ ${arch} = "aarch64" ]]; then
        virsh undefine --nvram "${domain}" || true
    else
        virsh undefine "${domain}" || true
    fi
}
...
# usage
undefine_vm "$DOMAIN_NAME" "$ARCH"
```

Reply:
While it's true that `--nvram` is only needed for EFI domains, which we currently only use on aarch64, it's perfectly safe to pass it unconditionally. If the NVRAM file doesn't exist, the flag will simply be ignored.

Reply:
Updated